linker/shared/src/main/scala/org/scalajs/linker/backend/wasmemitter/CoreWasmLib.scala

```diff
@@ -0,0 +1,2206 @@
+    val typeDataParam = fb.addParam("typeData", typeDataType)
+    val dimsParam = fb.addParam("dims", Int32)
+    fb.setResultType(objectVTableType)
+
+    val arrayTypeDataLocal = fb.addLocal("arrayTypeData", objectVTableType)
+
+    fb.loop() { loopLabel =>
+      fb.block(objectVTableType) { arrayOfIsNonNullLabel =>
+        // br_on_non_null $arrayOfIsNonNull typeData.arrayOf
+        fb += LocalGet(typeDataParam)
+        fb += StructGet(
+          genTypeID.typeData,
+          genFieldID.typeData.arrayOf
+        )
+        fb += BrOnNonNull(arrayOfIsNonNullLabel)
+
+        // <top-of-stack> := typeData ; for the <old typeData>.arrayOf := ... later on
+        fb += LocalGet(typeDataParam)
+
+        // typeData := new typeData(...)
+        fb += I32Const(0) // nameOffset
+        fb += I32Const(0) // nameSize
+        fb += I32Const(0) // nameStringIndex
+        fb += I32Const(KindArray) // kind = KindArray
+        fb += I32Const(0) // specialInstanceTypes = 0
+
+        // strictAncestors
+        for (strictAncestor <- strictAncestors)
+          fb += GlobalGet(genGlobalID.forVTable(strictAncestor))
+        fb += ArrayNewFixed(
+          genTypeID.typeDataArray,
+          strictAncestors.size
+        )
+
+        fb += LocalGet(typeDataParam) // componentType
+        fb += RefNull(HeapType.None) // name
+        fb += RefNull(HeapType.None) // classOf
+        fb += RefNull(HeapType.None) // arrayOf
+
+        // clone
+        fb.switch(RefType(genTypeID.cloneFunctionType)) { () =>
+          fb += LocalGet(typeDataParam)
+          fb += StructGet(genTypeID.typeData, genFieldID.typeData.kind)
+        }(
+          List(KindBoolean) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(BooleanRef))
+          },
+          List(KindChar) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(CharRef))
+          },
+          List(KindByte) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(ByteRef))
+          },
+          List(KindShort) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(ShortRef))
+          },
+          List(KindInt) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(IntRef))
+          },
+          List(KindLong) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(LongRef))
+          },
+          List(KindFloat) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(FloatRef))
+          },
+          List(KindDouble) -> { () =>
+            fb += ctx.refFuncWithDeclaration(genFunctionID.clone(DoubleRef))
+          }
+        ) { () =>
+          fb += ctx.refFuncWithDeclaration(
+            genFunctionID.clone(ClassRef(ObjectClass))
+          )
+        }
+
+        // isJSClassInstance
+        fb += RefNull(HeapType.NoFunc)
+
+        // reflectiveProxies
+        fb += ArrayNewFixed(genTypeID.reflectiveProxies, 0) // TODO
```

Comment:
Why the TODO? IIUC there cannot be a reflective proxy on an array? (It only has methods from Object, which never need a reflective proxy).

Reply:
Oh, right, I forgot to remove it :bow: